extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -197,19 +207,180 @@ const buildExpandedViewOverlay = element => htmlFor(element)`
 
 /**
  * Updates embed's corresponding <style> element with embedData.
+ * @param {!Element} target
  * @param {!Element} embedStyleEl
  * @param {!EmbedDataDef} embedData
  */
-function updateEmbedStyleEl(embedStyleEl, embedData) {
+function updateEmbedStyleEl(target, embedStyleEl, embedData) {
   const embedId = embedData.id;
+  embedStyleEl.textContent = `[${EMBED_ID_ATTRIBUTE_NAME}="${embedId}"]
+  ${buildStringStyleFromEl(target, embedData)}`;
+}
+
+/**
+ * Builds a string containing the corresponding style depending on the
+ * element.
+ * @param {!Element} target
+ * @param {!EmbedDataDef} embedData
+ * @return {string}
+ */
+function buildStringStyleFromEl(target, embedData) {
+  switch (target.tagName.toLowerCase()) {
+    case EXPANDABLE_COMPONENTS['amp-twitter'].selector:
+      return buildStringStyleForTweet(embedData);
+    default:
+      return buildDefaultStringStyle(embedData);
+  }
+}
+
+/**
+ * Builds string used in the <style> element for tweets.
```

Comment:
Nit: can you add that we ignore the height as the tweet is non deterministic?

Reply:
Done.